src/Microsoft.Kusto.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -270,6 +269,24 @@ public virtual async Task<ConnectionCompleteParams> Connect(ConnectParams connec
             return completeParams;
         }
 
+        internal void RefreshAzureToken(string ownerUri)
+        {
+            ConnectionInfo existingConnection = OwnerToConnectionMap[ownerUri];
+            
+            var requestMessage = new RequestSecurityTokenParams
+            {
+                Authority = existingConnection.ConnectionDetails.GetOptionValue("azureTenantId", string.Empty),
```

Comment:
You're probably going to want to also send the account ID as well so the ADS side doesn't have to ask the user which account it is.